controlplane/pkg/nsm/nsm.go

```diff
@@ -263,23 +259,31 @@ func (srv *networkServiceManager) request(ctx context.Context, request networkse
 
 		// 7.3 Destination context probably has been changed, so we need to update source context.
 		if err = srv.updateConnectionContext(cc.GetConnectionSource(), cc.GetConnectionDestination()); err != nil {
-			err = fmt.Errorf("NSM:(7.3-%v) Failed to update source connection context: %v", requestID, err)
-			srv.requestFailed(requestID, cc, existingCC, true, true)
-			return nil, err
+			err = fmt.Errorf("failed to update source connection context: %v", err)
```

Comment:
Why we can't call `srv.requestFailed` right here and avoid checking err on nil: https://github.com/Bolodya1997/networkservicemesh/blob/c5e2b8d12ee09c90ec9cb6703481de151002d229/controlplane/pkg/nsm/nsm.go#L267